ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/ListSourceExecutor.java

```diff
@@ -189,4 +238,16 @@ private static SourceDescription describeSource(
             q.getStatementString(), q.getSinkNames(), new EntityQueryId(q.getQueryId())))
         .collect(Collectors.toList());
   }
+
+  private static final class SourceDescriptionWithWarnings {
+    private final List<KsqlWarning> warnings;
+    private final SourceDescription description;
+
+    private SourceDescriptionWithWarnings(
+        final List<KsqlWarning> warnings,
+        final SourceDescription description) {
+      this.warnings = warnings;
```

Comment:
I know its internal, but I'd still null check these as it makes the code easier to grok, i.e. you know the field is never null. (Nulls are evil).